azcli_conformity.sh

```diff
@@ -0,0 +1,60 @@
+# Define Variables for the Configurations
+export APP_SECRET=$(uuidgen)
+
+# Ask User Input for the App Registration Name
+echo " "
+echo "Which name do you want to give to your App Registration?"
+IFS=$'\n'
+read -r APP_NAME
+echo " "
+echo "Hello, Thank you for trying this script out.
+We're configuring your Azure Account to work with Cloud One Conformity, this might take several minutes depending of how many subscriptions you have."
+
+# Create App Registration
+az ad app create --display-name "$APP_NAME" --password $APP_SECRET --credential-description secret --required-resource-accesses @manifest.json > /dev/null
+
+# Export App Registration ID and Active Directory ID
+export AD_ID=`az account show --query tenantId --output tsv`
+export APPID=`az ad app list --display-name "$APP_NAME" | grep appId | cut -d ":" -f2 | grep -o '".*"' | sed 's/^"\(.*\)".*/\1/'`
+
+# Create a Service Principal for the App Registration
+az ad sp create --id $APPID > /dev/null
+
+# Grant Admin Consent to application
+az ad app permission admin-consent --id $APPID
+
+# Export All Subscriptions ID's to an Array and Count to a Variable
+az account list --refresh > /dev/null
+export SUBSCRIPTIONS=(`az account list | grep id | cut -d ":" -f2 | grep -o '".*"' | sed 's/^"\(.*\)".*/\1/'`)
+NUM_SUBS=${#SUBSCRIPTIONS[@]}
+
+ # Loop to add each one of the Subscriptions
+ i=0
+ for (( i; i<=$NUM_SUBS; i++ ))
+ do  
+    az role definition create --role-definition '{ \
+      "Name": "Custom Role - Cloud One Conformity", \
```

Comment:
🔧  The custom role name is hard coded here and on line 49. It would be best if they were put into a variable to remove this duplication to make the code more maintainable.

Reply:
Agree

Reply:
Agree